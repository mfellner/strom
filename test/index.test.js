const Stream = require('../src');

describe('Stream', () => {
  test('require generator function', () => {
    expect(() => new Stream()).toThrowErrorMatchingSnapshot();
  });

  test('next', () => {
    const s = Stream.of(1);
    expect(s.next()).toBe(1);
  });

  test('isDone', () => {
    const s = Stream.of();
    expect(s.isDone()).toBe(false);
    expect(s.next()).toBeUndefined();
    expect(s.isDone()).toBe(true);
  });

  test('all operators', () => {
    const s = Stream.of(1, 2, 3, 4, 5, 6, 7, 8)
      .map(x => x * x)
      .filter(x => !!x)
      .flatMap(x => Stream.of(-x))
      .skip(4)
      .take(2);

    expect(s.toArray()).toEqual([-25, -36]);
  });
});
