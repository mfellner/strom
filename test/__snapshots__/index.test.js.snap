// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stream require generator function 1`] = `"new Stream() requires a generator function."`;
